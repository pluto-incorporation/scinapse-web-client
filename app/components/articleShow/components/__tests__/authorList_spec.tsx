jest.unmock("../authorList");

import * as React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import AuthorList, { IAuthorListProps } from "../authorList";

const mockUser = {
  nickName: "Jeffrey C. Lagarias",
  organization: "University of Michigan",
};

describe("<AuthorList /> component", () => {
  let authorListWrapper: ShallowWrapper<IAuthorListProps>;
  let mockAuthors;

  describe("when author is more than 3", () => {
    beforeEach(() => {
      mockAuthors = [mockUser, mockUser, mockUser, mockUser, mockUser];
      authorListWrapper = shallow(<AuthorList authors={mockAuthors} />);
    });

    it("should match snapshot", () => {
      expect(authorListWrapper.html()).toMatchSnapshot();
    });
  });

  describe("when author is less than 3", () => {
    beforeEach(() => {
      mockAuthors = [mockUser, mockUser];
      authorListWrapper = shallow(<AuthorList authors={mockAuthors} />);
    });

    it("should match snapshot", () => {
      expect(authorListWrapper.html()).toMatchSnapshot();
    });
  });
});
