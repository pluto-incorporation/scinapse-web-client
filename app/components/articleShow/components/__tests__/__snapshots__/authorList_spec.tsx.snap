// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorList /> component when author is less than 3 should match snapshot 1`] = `"<div class=\\"authorListWrapper\\"><span class=\\"authorItem\\"><div class=\\"contentWrapper\\"><div class=\\"authorName\\">pepsi</div><div class=\\"authorOrganization\\"></div></div></span><span class=\\"authorItem\\"><div class=\\"contentWrapper\\"><div class=\\"authorName\\">pepsi</div><div class=\\"authorOrganization\\"></div></div></span><span class=\\"authorItem\\"><div class=\\"contentWrapper\\"><div class=\\"authorName\\">pepsi</div><div class=\\"authorOrganization\\"></div></div></span></div>"`;

exports[`<AuthorList /> component when author is more than 3 should match snapshot 1`] = `"<div class=\\"authorListWrapper\\"><span class=\\"authorItem\\"><div class=\\"contentWrapper\\"><div class=\\"authorName\\">pepsi</div><div class=\\"authorOrganization\\"></div></div></span><span class=\\"authorItem\\"><div class=\\"contentWrapper\\"><div class=\\"authorName\\">pepsi</div><div class=\\"authorOrganization\\"></div></div></span><span class=\\"authorItem\\"><div class=\\"contentWrapper\\"><div class=\\"authorName\\">pepsi</div><div class=\\"authorOrganization\\"></div></div></span></div>"`;
